x/crosschain/keeper/cctx_orchestrator_validate_outbound.go

```diff
@@ -192,6 +193,18 @@ func (k Keeper) processFailedOutboundOnExternalChain(
 		if err != nil {
 			return err
 		}
+
+		// validate data of the revert outbound
+		err = k.validateZRC20Withdrawal(
```

Comment:
Need to be used in `processFailedOutboundV2` as well
In fact this function will be removed in the future

Reply:
It seems `processFailedOutboundV2` doesn't handle the case of reverting back to external chain. It currently invokes `AddRevertOutbound` and ends up with a call to `depositAndRevert` or `deposit` on the gateway ZEVM. Is there any subsequent flow (that I missed) to redirect the revert back to external chain (e.g., Bitcoin, Solana) and requires a TSS transaction?

Reply:
You're correct. We are still using `processFailedOutboundOnExternalChain` in V2. The other function is when the outbound fails on a connected chain (there is no longer external chain -> external chain cctx)